src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Text.cs

```diff
@@ -110,31 +110,9 @@ public override bool CheckTypes(CheckTypesContext checkTypesContext, TexlNode[]
             {
                 // Verify statically that the format string doesn't contain BOTH numeric and date/time
                 // format specifiers. If it does, that's an error according to Excel and our spec.
-
-                // But firstly skip any locale-prefix
-                if (formatArg.StartsWith("[$-"))
-                {
-                    var end = formatArg.IndexOf(']', 3);
-                    if (end > 0)
-                    {
-                        formatArg = formatArg.Substring(end + 1);
-                    }
-                }
-
-                var hasDateTimeFmt = formatArg.IndexOfAny(new char[] { 'm', 'd', 'y', 'h', 'H', 's', 'a', 'A', 'p', 'P' }) >= 0;
```

Comment:
@CarlosFigueiraMSFT  - 

Most of this code moved to the new helper, but this line is changed to include more characters:
            hasDateTimeFmt = formatArg.IndexOfAny(new char[] { 'm', 'M', 'd', 'D', 'y', 'Y', 'h', 'H', 's', 'S', 'a', 'A', 'p', 'P' }) >= 0;

Reply:
We should add as a test case ...
Text(1, "M#")

Reply:
This code also doesn't handle well scoped chars. For example, the following should be handled as a datetime format string:
`Text(1, "M""#""")`

Binder doesn't take into account that `#` is scoped. Also happening on make.test.
Are we looking to fix this before merging this PR?

Reply:
@MikeStall we talked about this validation before. I think we don't need to do it at compile time, leaving it to the runtime only - exactly because the validation is 1) incorrect (like in the example Anderson mentioned above) and 2) incomplete (for example, if the format comes from a variable).

Reply:
At a minimum - i.e., if we still want to keep this validation here - we must also do it at runtime, for expressions such as `With({format:"0.00 dd/mm/yyyy"},Text(123, format))` which are not evaluated at compile time.

Reply:
Consider this test case:
Text(123.466, "[$-en-US]$#0.0M") // $123.5M